### DISCUSSION TITLE
How to run Native Shell Programs?

### DISCUSSION THREAD

**Discussion by aborruso** [nushell/nushell#3487] [Q&A] [answered]
How to use in example `time`, inside nushell command?

Reading the [documentation](https://www.nushell.sh/cookbook/native_shell_programs.html) it should be

```
^time echo "ciao" | str find-replace "a" "aaa"
```

But I have 

```
error: Command not found
  ┌─ shell:2:1
  │
2 │ ^time echo "ciao" | str find-replace "a" "aaa"
  │ ^^^^^ command time not found
```

I'm using `nushell 0.30.0` inside debyan.

**Comment by sophiajt** [comment, marked as answer]
@aborruso `time` isn't a stand-alone command, but instead it's a command that's built into bash/zsh/fish/etc. You'd need to do something like `bash -c "time echo hello"`

If you're specifically looking to benchmark something, you can use the `benchmark` command that's part of Nushell.

**Reply by aborruso** [reply]
> If you're specifically looking to benchmark something, you can use the `benchmark` command that's part of Nushell.

In the doc I have

```
Runs a block and returns the time it took to execute it.

Usage:
  > benchmark <block> {flags}

Parameters:
  <block> the block to run and benchmark

Flags:
  -h, --help: Display this help message
  -p, --passthrough <block>: Display the benchmark results and pass through the block's output

Examples:
  Benchmarks a command within a block
  > benchmark { sleep 500ms }

  Benchmarks a command within a block and passes its output through
  > echo 45 | benchmark { sleep 500ms } --passthrough {}
```

But reading it I do not understand how to apply it to an easy hello world example.

If I run

```
ls | benchmark { sleep 500ms } --passthrough {}
```

I have nothing as output. But especially I do not understand the meaning of  `{ sleep 500ms } --passthrough {}`.

Why to benchmark something do I must insert a delay? How to view and access the block's output?

**Reply by sophiajt** [reply]
This is just an example, you can use any command or pipeline where it says `sleep 500ms`

For example, if you wanted to time how long it took to open a file called "file.json", you could do:

```
benchmark { open file.json }
```

**Reply by sophiajt** [reply]
If you want to view the block's output, you could use that second block. For example, something like:

```
benchmark { open file.json } --passthrough { each { echo $it } }

**Reply by aborruso** [reply]
Thank you very much
